File: test-ai-backend/src/main/java/com/testai/entity/Task.java
package com.testai.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 任务实体类
 * 对应数据库表 task
 */
@Data
@TableName("task")
public class Task implements Serializable {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /** 关联文档ID（单文档模式） */
    private Long documentId;

    /** 关联文档ID列表（多文档模式，逗号分隔） */
    private String documentIds;

    /** 文档类型 */
    private String documentType;

    /** 任务名称 */
    private String taskName;

    /** 任务号 */
    private String taskNo;

    /** 任务状态（0-待处理，1-进行中，2-已完成，3-失败） */
    private Integer taskStatus;

    /** 模板ID */
    private Long templateId;

    /** 配置信息（JSON格式） */
    private String config;

    /** 开始时间 */
    private LocalDateTime startTime;

    /** 结束时间 */
    private LocalDateTime endTime;

    /** 耗时（秒） */
    private Integer duration;

    /** 生成的测试用例数量 */
    private Integer resultCount;

    /** 质量评分 */
    private Double qualityScore;

    /** 反馈内容 */
    private String feedback;

    /** AI生成的响应内容 */
    private String aiResponse;

    /** 创建时间（自动填充） */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /** 更新时间（自动填充） */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

}

File: test-ai-frontend/src/api/task.ts
import request from '@/utils/request'

/**
 * 任务实体接口
 */
export interface Task {
  id: number                    // 任务ID
  documentId: number            // 关联文档ID（单文档模式）
  documentIds?: string          // 关联文档ID列表（多文档模式）
  documentType?: string         // 文档类型
  taskName: string              // 任务名称
  taskNo?: string               // 任务号
  taskStatus: number            // 任务状态（0-待处理，1-进行中，2-已完成，3-失败）
  templateId: number            // 模板ID
  config: string                // 配置信息（JSON格式）
  startTime: string             // 开始时间
  endTime: string               // 结束时间
  duration: number              // 耗时（秒）
  resultCount: number           // 生成的测试用例数量
  qualityScore: number          // 质量评分
  feedback: string              // 反馈内容
  aiResponse?: string           // AI生成的响应内容
  createTime: string            // 创建时间
  updateTime: string            // 更新时间
}

/**
 * 任务创建请求接口
 */
export interface TaskCreateRequest {
  documentId: number            // 文档ID
  taskName: string              // 任务名称
  templateId?: number           // 模板ID（可选）
  config?: string               // 配置信息（可选）
}

/**
 * 任务反馈请求接口
 */
export interface TaskFeedbackRequest {
  rating?: number               // 评分（可选）
  feedback?: string             // 反馈内容（可选）
  adoptionRate?: number         // 采用率（可选）
}

export interface AiConfig {
  platform: string              // AI平台（aliyun/openai/gemini）
  apiUrl: string                // API地址
  model: string                 // 模型名称
  apiKey: string                // API密钥
}

/**
 * 测试用例生成请求接口
 */
export interface GenerateTestCaseRequest {
  documentId?: number           // 单文档ID（单文档模式，可选）
  documentIds?: number[]        // 多文档ID列表（多文档模式，可选）
  taskName: string              // 任务名称
  taskNo: string                // 任务号
  templateId?: number           // 模板ID（可选）
  aiConfig: AiConfig            // AI配置信息
}

export function createTask(data: TaskCreateRequest) {
  return request.post<Task>('/api/tasks/create', data)
}

export function getTask(id: number) {
  return request.get<Task>(`/api/v1/tasks/${id}`)
}

export function deleteTask(id: number) {
  return request.delete(`/api/tasks/${id}`)
}
/**
 * 生成测试用例
 * 
 * @param data 生成请求参数
 * @returns 返回任务ID
 */
export function exportAiResponse(id: number) {
  return request.get(`/api/tasks/${id}/export-ai-response`, {
    responseType: 'blob'
  })
}

export function generateTestCases(data: GenerateTestCaseRequest) {
  return request.post<{ taskId: number }>('/api/v1/tasks/generate', data)
}

export function getTaskStatistics() {
  return request.get('/api/v1/tasks/statistics')
}

export function getTaskList(data: any) {
  return request.post('/api/v1/tasks/list', data)
}

export function getTaskDetail(id: number) {
  return request.get(`/api/v1/tasks/detail/${id}`)
}

export function downloadTestCase(id: number) {
  return request.get(`/api/v1/tasks/${id}/download`, {
    responseType: 'blob'
  })
}

export function submitTaskFeedback(id: number, feedback: string) {
  return request.post(`/api/v1/tasks/${id}/feedback`, { feedback })
}
